### ISSUE TITLE
Code for IBM i will not function correctly until the temporary library has been corrected in the settings.

### ISSUE THREAD

**Issue by Fdeuabbh** [codefori/vscode-ibmi#2363]
Code for i is connected to the server successfully, but when i've got the message 
"Code for IBM i will not function correctly until the temporary library has been corrected in the settings." i could not access to the libraries. 

Is there any solution ?

![image](https://github.com/user-attachments/assets/e7b053c9-9a55-41c6-9532-9de73cac4000)

**Comment by sebjulliand** [comment]
Hi @Fdeuabbh ,
click on the `Open Settings` button and then change these values
![image](https://github.com/user-attachments/assets/7a62aa27-7c88-481a-bee7-16dde6b3218c)

You must use a library and an IFS folder your IBM i profile can write into.

**Comment by sebjulliand** [comment]
Hi @kziem ! /QOpenSys/pkgs/bin being not found sounds very concerning 😅

Could you check if it actually exists or not? If it exists, is it accessible to you?

**Comment by kziem** [comment]
There is no folder /QOpenSys/pkgs on this system 😕
I've got access to all subfolders and files in /QOpenSys
maybe its moved somewhere else, can i find the files that are supposed to be in it?

**Comment by sebjulliand** [comment]
No worries. You won't find its content anywhere else 🥲
This pkgs subfolder contains the open source softwares installed on IBM i. So first, try installing the OSS and yum: https://ibmi-oss-docs.readthedocs.io/en/latest/yum/README.html
Then make sure bash is installed and it's your default shell. (Check the code for i  documation as well as the site linked above)
Once you're all set, Code for i should run just fine!

**Comment by kziem** [comment]
Thanks for your reply 🙂
Sadly I cannot really install anything on that system, but I found sameish files in /Qopensys/bin folder, can I somehow change config of extension to use that instead?

**Comment by sebjulliand** [comment]
No, unfortunately you'll be stuck until you can get a friendly sys admin to install these tools for you.

/QOpensys/bin content is a different set of binaries, don't mess with QOpensys' content in any way 😅
